addons/account/models/res_partner_bank.py

```diff
@@ -63,6 +64,21 @@ def _check_allow_out_payment(self):
                 if not self.env.user.has_group('account.group_validate_bank_account'):
                     raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
 
+    @api.depends('acc_number')
+    def _compute_alert_duplicate_bank_account(self):
+        """Add a warning if the bank account is already in use elsewhere"""
+        acc_numbers = [bank.acc_number for bank in self if bank.acc_number]
+        duplicates = self.env['res.partner.bank'].search([('acc_number', 'in', acc_numbers)])
+        for bank in self:
+            bank.duplicate_account_warning = False
+            if bank.acc_number:
+                duplicate = next((dup for dup in duplicates if dup.acc_number == bank.acc_number and bank.id != dup.id), None)
+                if duplicate:
```

Comment:
```suggestion
            if duplicate := duplicates.get(bank.acc_number):
                if bank.id != duplicate.id:
```
That's a weird use of `next()` and the whole thing seems overly complicated. This is easier to understand.

What do you think ?

Reply:
Agree for the "next", I don't really like the staircase of ifs... It *is* clearer though, so I did the change.